Add automatic linux/mac switch to Makefile

neonsoftware opened the pull request:
Hi @markuswustenberg,

Just tested on my Linux. Let me know if ok also for Mac.

Have a nice day,

markuswustenberg commented:
@neonsoftware thank you for the contribution!

I'd rather not explicitly not support Windows. But I don't have access to a Windows box to test through something like WSL. Could you delete the last conditional block?

markuswustenberg commented:
@neonsoftware bonus points for figuring out Linux on ARM64 support as well. 😊

neonsoftware commented:
Hi, 

### updates

- added arm64 support (points unlocked ?)
- added windows support, see below for test

### windows specific

- windows required change a specific tailwind installation, see Makefile
- windows server is slower to start, it required an incremental startup wait in tests

### ci
 in order to test the Makefile-based flow on all supported platforms I have added a CI worflow

### question
The project has a nice makefile, although ci.yml does not use it, instead it rewrites the go commands. Is there a specific reason ?
Because instead of adding a new worflow, i could, of course, have **added** the platform matrix to your workflow ci.yml, but I thought that maybe you want that flow Makefile-free...

Let me know, 

Have a nice day

markuswustenberg commented:
Hmm. I'll be honest, this is a bit too much complexity for downloading a CLI tool for the right platform for my taste, and not something I'd like to have to maintain going forward. IMO, users of the template can just change the platform-specific stuff in their own repo after forking.

It's not that I don't want to support Windows users, but I am also not willing to learn the specifics of a platform I haven't personally used for two decades.

So while I appreciate that you want to make it work on all platforms, this unfortunately isn't something I want to merge. Sorry for your trouble, and thank you for your time!

neonsoftware commented:
Hi @markuswustenberg,

no problem at all.

You can close the issue.

Have a nice day
